### PR TITLE
AKS Workflow: step 1/8

### PR DESCRIPTION
**Pull request by bear454**

### CODE REVIEW

**Review by jesusbv — APPROVED** [review]

File: engines/azure/app/models/azure/resource_group.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      self.id = response['id']
+      self.refresh()
+    end
+
+    def ready!
+      self.wait_until('Succeeded')
+    end
+
+    def azure_destroy()
+      @cli.destroy_resource_group(name: @name)
```
Comment (jesusbv) [inline engines/azure/app/models/azure/resource_group.rb:19]: do we not want to generate script to download for the user with the command ?

File: engines/rancher_on_eks/app/models/rancher_on_eks/deployment.rb [hunk 2]
```diff
@@ -30,7 +30,7 @@ def create_steps!
       Step.create!(
         rank: 6,
         duration: 16,
-        action: 'Route public subnets through Internet Gateway'
+        action: 'Route public subnets through the Internet Gateway'
```
Comment (jesusbv) [inline engines/rancher_on_eks/app/models/rancher_on_eks/deployment.rb:33]: this file changes should be a different PR (EKS changes mixed with AKS ones)

File: engines/azure/app/models/azure/resource_group.rb (hunk 1 above)
Comment (rjschwei) [reply engines/azure/app/models/azure/resource_group.rb:19]: A one liner is IMHO not worth doing anything extra.
